nvflare/app_common/widgets/log_streaming.py

```diff
@@ -57,6 +96,8 @@ def save(self, fl_ctx: FLContext, shareable: Shareable, record_origin: str):
                 handler = self.handlers.get(record_origin)
```

Comment:
I think the lock should be placed before self.handlers.get().

Reply:
Yes, it should.
I've fixed this issue.